### PR TITLE
Emcs 633 workitem

### PR DESCRIPTION
**Pull request by Froom2**
Adds the basic workitem and scheduler for NRS submissions. No wiring.

### CODE REVIEW

File: app/uk/gov/hmrc/excisemovementcontrolsystemapi/scheduling/NrsSubmissionScheduler.scala [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    val correlationId           = correlationIdService.generateCorrelationId()
+    val result: Future[Boolean] = nrsWorkItemRepository
+      .pullOutstanding(now, now)
+      .flatMap {
+        case None           => Future.successful(true)
+        case Some(workItem) =>
+          nrsConnector
+            .sendToNrs(workItem.item.payload, correlationId)(HeaderCarrier())
+            .flatMap {
+              case NonRepudiationSubmissionAccepted(_)       =>
```
Comment (RobertLaverick) [inline app/uk/gov/hmrc/excisemovementcontrolsystemapi/scheduling/NrsSubmissionScheduler.scala:52]: Correct me if I'm wrong but this type is a wrapper around an Id? Why not just called it NRSSubmissionId?

File: test/uk/gov/hmrc/excisemovementcontrolsystemapi/scheduling/NrsSubmissionSchedulerSpec.scala [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+    }
+    "use initial delay from configuration" in {
+      nrsSubmissionScheduler.initialDelay mustBe FiniteDuration(1, "minutes")
+    }
+    "use interval from configuration" in {
+      nrsSubmissionScheduler.interval mustBe FiniteDuration(1, "minute")
+    }
+  }
+
+  ".execute" - {
```
Comment (RobertLaverick) [inline test/uk/gov/hmrc/excisemovementcontrolsystemapi/scheduling/NrsSubmissionSchedulerSpec.scala:110]: Looks good, my only concern is there isn't any test around retrying a failed work item.

We should have a scenario of 'this failed too many times, pagerduty' which can leverage failureCount

**Review by RobertLaverick — CHANGES_REQUESTED** [review]

File: app/uk/gov/hmrc/excisemovementcontrolsystemapi/scheduling/NrsSubmissionScheduler.scala (hunk 1 above)
Comment (Froom2) [reply app/uk/gov/hmrc/excisemovementcontrolsystemapi/scheduling/NrsSubmissionScheduler.scala:52]: This is type is probably going to get deleted in the next PR :) We aren't changing existing functionality in this PR at all

File: test/uk/gov/hmrc/excisemovementcontrolsystemapi/scheduling/NrsSubmissionSchedulerSpec.scala (hunk 2 above)
Comment (Froom2) [reply test/uk/gov/hmrc/excisemovementcontrolsystemapi/scheduling/NrsSubmissionSchedulerSpec.scala:110]: That will be added in the next PR :) but yes, agreed.

File: app/uk/gov/hmrc/excisemovementcontrolsystemapi/scheduling/NrsSubmissionScheduler.scala (hunk 1 above)
Comment (RobertLaverick) [reply app/uk/gov/hmrc/excisemovementcontrolsystemapi/scheduling/NrsSubmissionScheduler.scala:52]: Sounds reasonabe to me

File: test/uk/gov/hmrc/excisemovementcontrolsystemapi/scheduling/NrsSubmissionSchedulerSpec.scala (hunk 2 above)
Comment (RobertLaverick) [reply test/uk/gov/hmrc/excisemovementcontrolsystemapi/scheduling/NrsSubmissionSchedulerSpec.scala:110]: Right, I'll consider it resolved then

**Review by RobertLaverick — APPROVED** [review]

**Pull request merged**